tests/Feature/OrderTest.php

```diff
@@ -211,6 +211,86 @@ public function test_create_order_validation_fails()
+            'address_invoice_id' => $this->addresses[0]->id,
+            'address_delivery_id' => $this->addresses[1]->id,
+            'client_id' => $this->clients[0]->id,
+            'language_id' => $this->languages[0]->id,
+            'order_type_id' => $this->orderTypes[0]->id,
+            'payment_type_id' => $this->paymentTypes[0]->id,
+            'price_list_id' => $this->priceLists[0]->id,
+            'address_delivery' => [
+                'company' => 'test-company',
+            ],
+            'payment_target' => rand(10, 20),
+            'payment_discount_target' => rand(3, 5),
+            'payment_discount_percent' => rand(1, 10) / 100,
+            'payment_reminder_days_1' => rand(1, 10),
+            'payment_reminder_days_2' => rand(1, 10),
+            'payment_reminder_days_3' => rand(1, 10),
+            'payment_texts' => [Str::random(), Str::random(), Str::random()],
+            'order_date' => date('Y-m-d', strtotime('+1 day')),
+        ];
+
+        $this->user->givePermissionTo($this->permissions['create']);
+        Sanctum::actingAs($this->user, ['user']);
+
+        $response = $this->actingAs($this->user)->post('/api/orders', $order);
+        $response->assertStatus(201);
+
+        $responseOrder = json_decode($response->getContent())->data;
+        $dbOrder = Order::query()
+            ->whereKey($responseOrder->id)
+            ->first();
+
+        $this->assertEquals($order['address_delivery']['company'], $dbOrder->address_delivery['company']);
+        $this->assertNull($dbOrder->address_delivery_id);
+    }
+
+    public function test_update_order_address_delivery_validation_fails()
+    {
+        $order = [
+            'id' => $this->orders[0]->id,
+            'address_delivery' => [
+                'zip' => 12345,
+            ],
+        ];
+
+        $this->user->givePermissionTo($this->permissions['update']);
+        Sanctum::actingAs($this->user, ['user']);
+
+        $response = $this->actingAs($this->user)->put('/api/orders', $order);
+        $response->assertStatus(422);
+
+        $response->assertJsonValidationErrorFor('address_delivery.zip');
+    }
+
+    public function test_update_order_address_delivery()
+    {
+        $order = [
+            'id' => $this->orders[0]->id,
+            'address_delivery' => [
+                'company' => 'test company',
+            ],
+        ];
+
+        $this->user->givePermissionTo($this->permissions['update']);
+        Sanctum::actingAs($this->user, ['user']);
+
+        $response = $this->actingAs($this->user)->put('/api/orders', $order);
+        $response->assertStatus(200);
+
+        $responseOrder = json_decode($response->getContent())->data;
+        $dbOrder = Order::query()
+            ->whereKey($responseOrder->id)
+            ->first();
+
+        $this->assertEquals($order['address_delivery']['company'], $dbOrder->address_delivery['company']);
+        $this->assertNull($dbOrder->address_delivery_id);
```

Comment:
ideally you would want to check if the original order maintained the same, this is super minimalist test